warehouse/query-core/src/main/java/datawave/query/iterator/logic/KeywordExtractingIterator.java

```diff
@@ -0,0 +1,388 @@
+                    .singleton(new ArrayByteSequence(Constants.D_COLUMN_FAMILY.getBytes()));
+
+    public static final String MIN_NGRAMS = "min.ngram.count";
+    public static final String MAX_NGRAMS = "max.ngram.count";
+    public static final String MAX_KEYWORDS = "max.keyword.count";
+    public static final String MAX_SCORE = "max.score";
+    public static final String MAX_CONTENT_CHARS = "max.content.chars";
+    public static final String VIEW_NAMES = "view.names";
+    public static final String ONLY_SPECIFIED_VIEWS = "only.specified.views";
+
+    public static final Map<String,String> defaultMapOptions;
+
+    static {
+        defaultMapOptions = Maps.newHashMap();
+        defaultMapOptions.put(MIN_NGRAMS, "minimum number of words (ngrams) per keyword");
+        defaultMapOptions.put(MAX_NGRAMS, "maximum number of words (ngrams) per keyword");
+        defaultMapOptions.put(MAX_KEYWORDS, "maximum number of keywords to extract");
+        defaultMapOptions.put(MAX_SCORE, "max keyword score allowed (smaller scores are better)");
+        defaultMapOptions.put(MAX_CONTENT_CHARS, "max number of input characters to process");
+        defaultMapOptions.put(VIEW_NAMES, "a comma separated list of views to extract keywords from, in priority order");
+        defaultMapOptions.put(ONLY_SPECIFIED_VIEWS,
+                        "only extract keywords from the views specified in the iterator options, " + "not from the default list configured in the iterator");
+    }
+
+    private static final int MAX_CONTENT_ONE_HUNDRED_MB = 100 * 1024 * 1024;
+
+    /**
+     * A list of views we'll attempt to use for content from which to extract keywords, ordered based on preference.
+     */
+    protected final List<String> preferredViews = new ArrayList<>();
+
+    /**
+     * if we will only look at the view names specified in the query parameters instead of merging those views with the list of views in the configuration and
+     * then using the resulting list.
+     */
+    protected boolean onlySpecifiedViews;
+
+    /** the underlying source */
+    protected SortedKeyValueIterator<Key,Value> source;
+
+    /** the options this iterator is configured to use */
+    protected Map<String,String> iteratorOptions;
+
+    /** The specified dt/uid column families */
+    protected SortedSet<String> columnFamilies;
+
+    /** inclusive or exclusive dt/uid column families */
+    protected boolean inclusive;
+
+    /** the underlying D column scan range */
+    protected Range scanRange;
+
+    /** the top key */
+    protected Key tk;
+
+    /** the top value */
+    protected Value tv;
+
+    @Override
+    public boolean hasTop() {
+        return tk != null;
+    }
+
+    @Override
+    public SortedKeyValueIterator<Key,Value> deepCopy(IteratorEnvironment env) {
+        KeywordExtractingIterator it = new KeywordExtractingIterator();
+        it.source = source.deepCopy(env);
+        return it;
+    }
```

Comment:
this should copy everything that was set or configured in init() into the new iterator. I think there is more to do here.